kyo-reactive-streams/shared/src/main/scala/kyo/interop/flow/StreamSubscription.scala

```diff
@@ -0,0 +1,154 @@
+package kyo.interop.flow
+
+import StreamSubscription.*
+import java.util.concurrent.Flow.*
+import kyo.*
+import kyo.interop.flow.*
+import kyo.kernel.ArrowEffect
+import kyo.kernel.Boundary
+import kyo.kernel.Safepoint
+import kyo.scheduler.IOPromise
+import kyo.scheduler.IOTask
+
+final private[kyo] class StreamSubscription[V, Ctx](
+    private val stream: Stream[V, Ctx],
+    subscriber: Subscriber[? >: V]
+)(
+    using
+    AllowUnsafe,
+    Frame
+) extends Subscription:
+
+    private val requestChannel = Channel.Unsafe.init[Long](Int.MaxValue)
+
+    override def request(n: Long): Unit =
+        if n <= 0 then subscriber.onError(new IllegalArgumentException("non-positive subscription request"))
+        discard(requestChannel.offer(n))
+    end request
+
+    override def cancel(): Unit =
+        given Frame = Frame.internal
+        discard(requestChannel.close())
+    end cancel
+
+    private[interop] inline def subscribe(using Frame): Unit < IO = IO(subscriber.onSubscribe(this))
+
+    private[interop] def poll(using Tag[Poll[Chunk[V]]], Frame): StreamFinishState < (Async & Poll[Chunk[V]]) =
```

Comment:
I tried to convert this into `channel.stream()`, but this provides more controls over the computation